### ISSUE TITLE
Breaking change with "adguard/adguardhome" latest image

### ISSUE THREAD

**Issue by timk153** [muhlba91/external-dns-provider-adguard#37]
**TLDR**
adguard home v0.107.43 (latest) introduces breaking changes.

v0.107.42 and 41 not tested.

**Reproduction steps:**

- Use external-dns: 0.14.0-debian-11-r2 (installed by bitnami chart with default values)
- Use external-dns-provider-adguard: v3.1.0
- Use adguard home: [v0.107.43](https://github.com/AdguardTeam/AdGuardHome/releases/tag/v0.107.43)

**Problem description**
The custom filtering rules section within Adguard:

- Rules syntax appears as JSON formatted text string. From having had this operational I don't think this is intended behaviour.
- Additional '/' characters get added for each execution of the hook
- Hook always detects changes to be made, even if none are required
- DNS resolution for entries fails

**Workaround**
Have hard coded adguard home to v0.107.40 which would have been current at the time of the last release.
This version behaves correctly.

**Comment by muhlba91** [comment]
thanks for raising this issue!

could you also post a (redacted) snapshot of the rules you see in adguard?

(fyi, i'll have a look at it in the next days since i'm coming back from vacation tomorrow.)

**Comment by muhlba91** [comment]
i just checked my setup and all looks fine to me with:
adguard home: v0.107.43
external-dns: v0.14.0
external-dns-provider-adguard: v3.1.0

**Comment by timk153** [comment]
Just attempted with a quick version number change. 
I'm unable to replicate it.

I'll have another try shortly and will update the comment accordingly.

**Comment by timk153** [comment]
Unable to replicate
